core/model/modx/processors/search/search.class.php

```diff
@@ -18,40 +33,584 @@ class modSearchProcessor extends modProcessor
+        );
+
+        foreach ($collection as $record) {
+
+            if ($element == 'ss') {
+                $id = $record->get('key');
+                $name = $this->modx->lexicon('setting_' . $record->get('key'));
+                $description = '';
+                $content = $record->get('value');
+            } else {
+                $id = $record->get('id');
+                $name = $record->get($this->elements[$element]['name']);
+                $description = $record->get($this->elements[$element]['description']);
+                $content = $record->get($field);
+            }
+
+            $icon = $this->elements[$element]['icon'];
+            $action = $this->elements[$element]['action'];
+            $type = $this->elements[$element]['type'];
+
+
+            if (in_array($field, array('plugincode', 'snippet', 'content', 'introtext'))) {
+                // find line where $search can be found
+                $lines = explode("\n", $content);
+                $content = '';
+                $search = str_replace('%', '', $search);
+
+                $i = 0;
+                foreach ($lines as $linenumber => $line) {
+                    $i++;
+                    if (is_numeric(stripos($line, $search))) {
+                        $line = htmlspecialchars($line);
+                        $linenumber = str_pad($linenumber + 1, 3, '0', STR_PAD_LEFT);
+                        $line = str_replace($search, '<strong>' . $search . '</strong>', $line);
+                        $content .= "<pre class='line'><span class='number'>$linenumber</span><code>$line</code></pre>";
+                    }
                 }
+            } else {
+                $content = str_replace($search, "<strong>$search</strong>", $content);
             }
+
+            $this->results[] = array(
+                'uberbar_header' => $this->modx->lexicon('uberbar_header'),
+                'msg' => '',
+                'options' => '',
+                'header' => '',
+                'id' => $id,
+                'name' => $name,
+                'description' => $description,
+                'content' => $content,
+                'icon' => $icon,
+                '_action' => $action . '&id=' . $id,
+                'type' => $type,
+                'current' => $start++
+            );
         }
+    }
 
-        return $this->outputArray($this->results);
+    /**
+     * Validates the uberbar to check for permissions, allowed elements and fields etc.
+     *
+     * @param $query
+     * @return bool
+     */
+    public function validateUberbar($query)
+    {
+        $mode = $this->modx->getOption('uberbar_mode');
+        $count = count($query);
+        $element = isset($query[1]) ? $query[1] : false;
+        $field = isset($query[2]) ? $query[2] : false;
+
+        // Element related stuff...
+        if ($count > 2) {
+            /* Validate elements */
+            if ($element != '*' && !array_key_exists($element, $this->elements)) {
```

Comment:
While the idea seems good, i'd say that won't work : 

1. possibly allow "overrides" (as in, what if i register another element with the `r` trigger)
2. i would have no control over the action, since the query just append an `&id=` to `$this->elements['trigger']['action']` (see l.310-320 & l.357). objects might have a different PK.